Warning in `Manifold::get_new_point()`

Compiling in release mode gives me the following `maybe-uninitialized` warning. I am using `gcc 15.1.1` and `boost 1.88.0`.
```
Building CXX object source/grid/CMakeFiles/object_grid_release.dir/manifold.cc.o
/home/marc/work/codes/dealii/dealii/source/grid/manifold.cc: In member function ‘dealii::Point<spacedim> dealii::Manifold<dim, spacedim>::get_new_point(const dealii::ArrayView<const dealii::Point<spacedim> >&, const dealii::ArrayView<const double>&) const [with int dim = 1; int spacedim = 1]’:
/home/marc/work/codes/dealii/dealii/source/grid/manifold.cc:96:41: warning: ‘*(unsigned int*)((char*)&permutation + offsetof(boost::container::small_vector<unsigned int, 100, void, void>,boost::container::small_vector<unsigned int, 100, void, void>::<unnamed>.boost::container::small_vector_storage<unsigned int, 100, 4>::m_storage))’ may be used uninitialized [-Wmaybe-uninitialized]
   96 |   Point<spacedim> p = surrounding_points[permutation[0]];
      |                       ~~~~~~~~~~~~~~~~~~^
/home/marc/work/codes/dealii/dealii/source/grid/manifold.cc:87:53: note: ‘permutation’ declared here
   87 |   boost::container::small_vector<unsigned int, 100> permutation(n_points);
      |                                                     ^~~~~~~~~~~
```
For context, the permutation variable, which is a `boost::container::small_vector`, has been introduced with #5024.

All my attempts correcting it have failed so far. Do you have any ideas?

--- a/source/grid/manifold.cc
+++ b/source/grid/manifold.cc
@@ -93,9 +93,10 @@ Manifold<dim, spacedim>::get_new_point(
             });
 
   // Now loop over points in the order of their associated weight
+  DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
   Point<spacedim> p = surrounding_points[permutation[0]];
-  double          w = weights[permutation[0]];
-
+  DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
+  double w = weights[permutation[0]];
   for (unsigned int i = 1; i < n_points; ++i)
     {
       double weight = 0.0;

